### ISSUE TITLE
Objectives for Gunsmith style quests

### ISSUE THREAD

**Issue by Ygnas** [TarkovTracker/tarkovdata#122]
I guess you moved to using `id`'s and doing it by #2 with durability, ergonomics etc. doesn't look that great.
So what you think about this way to do `objectives`?
```
{
        "id": 82,
        "require": {
            "level": 2,
            "quests": []
        },
        "giver": 4,
        "turnin": 4,
        "title": "Gunsmith - Part 1",
        "locales": {
            "en": "Gunsmith - Part 1",
            "ru": "Оружейник. Часть 1",
            "cs": "Puškař - Část 1"
        },
        "wiki": "https://escapefromtarkov.fandom.com/wiki/Gunsmith_-_Part_1",
        "exp": 5400,
        "unlocks": [],
        "reputation": [
            {
                "trader": 4,
                "rep": 0.02
            }
        ],
        "objectives": [
            {
                "type": "build",
                "target": "584148f2245977598f1ad387",
                "number": 1,
                "notes": "This build is based on a standard MP-133 from Jaeger LL1",
                "parts": [
                    {
                        "id": "56083a334bdc2dc8488b4571",
                        "name": "MP-133/153 plastic pistol grip"
                    },
                    {
                        "id": "55d48ebc4bdc2d8c2f8b456c",
                        "name": "Delta-tek Sprut mount for pump-action shotguns"
                    },
                    {
                        "id": "5cc9c20cd7f00c001336c65d",
                        "name": "NcSTAR Tactical blue laser LAM-Module"
                    },
                    {
                        "id": "560838c94bdc2d77798b4569",
                        "name": "Remington Tactical Choke 12ga"
                    }
                ],
                "id": 422
            }
        ],
        "gameId": "5ac23c6186f7741247042bad"
    },
```

**Comment by thaddeus** [comment]
I have mixed feelings about suggesting specific parts. On the one hand, its the easiest for the users, on the other hand, its not *really* encapsulating the information accurately.

My latest thoughts are I would like the build objectives to be the requirements for the quest (ergo, recoil, specific parts, size, etc) but then have a link to something like https://tarkov-gunsmith.com/ ideally if it could be a folder of potential builds that currently fulfill the requirements. I think that link would actually best fit in a sister-project to tarkovdata which I envision (something that links content creator videos/guides and other resources to tarkovdata ids)

I don't know what the 'right' solution for this is, would love to have your thoughts and maybe @kokarn or @Razzmatazzz's as well.

**Comment by Razzmatazzz** [comment]
I agree with the approach to set out the overall requirements (including specific parts where applicable) but avoid including full builds.

One reason I don't like the idea of including full builds is that there's more than one way to complete most of these quests. Now the the flea unlocks at level 20, it's much more likely that the first couple Gunsmith quests will be completed from low-level trader parts and parts actually found in raid. Including a full build in the quest data implies that's the only way to complete the quest when there are potentially other avenues available.

Another reason is that the attributes of attachments can and do change, and that can invalidate builds even if the quest requirements don't change. That creates significant maintenance overhead if full builds are included.

**Comment by Ygnas** [comment]
Of course everything is less or equal except magazine.
By the wiki any tactical device should do. So not sure if should specify `laser` cause that's easiest I guess.
```
"objectives": [
            {
                "type": "build",
                "target": "584148f2245977598f1ad387",
                "number": 1,
                "with": [
                    {
                        "type": "ergonomics",
                        "value": 47
                    },
                    {
                        "type": "recoil",
                        "value": 950
                    },
                    {
                        "type": "cells",
                        "value": 4
                    },
                    {
                        "type": "attachment",
                        "name": "Extended magazine",
                        "value": 6
                    },
                    {
                        "type": "attachment",
                        "name": "Tactical device"
                    }
                ],
                "id": 422
            }
        ],
```
What you think this way then? And if you have something in mind say it. Maybe I should just do it by #2, if you think that's the best.

**Comment by thaddeus** [comment]
@Ygnas Sorry, meant to reply to this the other day but got lost in the process.

I like this format. A couple tweaks I personally would prefer:
- Make the target the base item ID instead of a item_preset (since we're not going the route of building a specific gun with specific parts)
- Change the `type` for things like `recoil` and `ergonomics` to `stat` (like in #2 ) and add `name` for the `recoil`, `ergonomics`, etc. I think that's valuable information to help make it easy for tools like TarkovTracker to consume. For example, in a list of quests, I probably would just want to display "custom stats needed: ergonomics, recoil, cells" and then show the exact values in a detail view. This helps categorize that.
- I like the attachment type with category name of the attachment, and then minimum value
- Stat values should probably have the operator like in the #2 example, some cases we want above certain value (ergonomics >40, sighting range >=1500) and others we want below (recoil < 250).

Thanks for helping figure this out!

**Comment by thaddeus** [comment]
Resolved by #128
